interfaces/PA-App/src/app/services/logging.service.ts

```diff
@@ -0,0 +1,99 @@
+import { Injectable } from '@angular/core';
+import { ApplicationInsights } from '@microsoft/applicationinsights-web';
+import { SeverityLevel } from 'src/app/models/severity-level.model';
+import { environment } from 'src/environments/environment';
+
+@Injectable()
+export class LoggingService {
+  appInsights: ApplicationInsights;
+  constructor() {
+    if (environment.ai_ikey && environment.ai_endpoint) {
+      this.appInsights = new ApplicationInsights({
+        config: {
+          connectionString:
+            'InstrumentationKey=' +
+            environment.ai_ikey +
+            ';IngestionEndpoint=' +
+            environment.ai_endpoint,
+          instrumentationKey: environment.ai_ikey,
+          enableAutoRouteTracking: true,
+        },
+      });
+
+      this.appInsights.loadAppInsights();
+      this.appInsights.addTelemetryInitializer(this.addTelemetryProcessor);
+    }
+  }
+
+  addTelemetryProcessor(envelope: any) {
+    const baseData = envelope.baseData;
+
+    // filter audio files
+    if (
+      baseData.responseCode === 404 &&
+      baseData.name.match(
+        '/assets/i18n/(en|de)/[a-z.-]*.(en|ny_MW|tuv_KE|saq_KE|om_ET).(webm|mp3)',
+      )
+    ) {
+      return false;
+    }
+
+    // filter/sample data
+
+    return false;
+  }
+
+  logPageView(name?: string, url?: string) {
+    if (environment.ai_ikey && environment.ai_endpoint) {
```

Comment:
I think this repeated condition could be put in a separate method...
Or maybe even stop at the "constructor"-level, as nothing will work without these values...